README.md

```diff
@@ -19,8 +19,240 @@
+3.
+ <discover-state> (background: false) (weak: false)
+4.
+ <mount-root> (background: false) (weak: false)
+5.
+ <mount-oem> (background: false) (weak: false)
+6.
+ <rootfs-hook> (background: false) (weak: false)
+7.
+ <load-config> (background: false) (weak: false)
+8.
+ <custom-mount> (background: false) (weak: false)
+ <overlay-mount> (background: false) (weak: false)
+9.
+ <mount-bind> (background: false) (weak: false)
+10.
+ <write-fstab> (background: false) (weak: true)
+```
+
+As shown in the DAG, the steps are in order and that shows their dependencies, i.e. `mount-root` depends on `discover-state`and that is why it's just below it.
+It won't run until the previous step has completed **without errors**.
+There is also the `weak` value which indicates that this step has weak dependencies. It will run even if its dependencies failed, instead of refusing to run.
+
+
+
+### Steps explained
+
+ - `mount-state`: Will mount the `COS_STATE` partition under `/run/initramfs/cos-state`
+ - `mount-tmpfs`: Will mount `/tmp` 
+ - `create-sentinel`: Will create the sentinel file identifying the boot mode (`active_mode`, `passive_mode`, `recovery_mode` or `live_mode`) under `/run/cos/`
+ - `mount-base-overlay`: Will mount the base overlay under `/run/overlay`
+ - `discover-state`: Will find the correct image under `/run/initramfs/cos-state` and mount it as a loop device
+ - `mount-root`: Will mount the `/dev/disk/by-label/$LABEL` device under the sysroot (Usually `/sysroot`). This label is set in grub depending on the selected entry, as part of the cmdline (i.e. `root=LABEL=COS_ACTIVE`) 
+ - `mount-oem`: Will **try** to mount the oem label device under `/sysroot/oem`. This label is set in grub by default (`rd.cos.oemlabel=COS_OEM`) but also on the default `cos-layout.env` file with Kairos. This partition is not mandatory so It's allowed to fail.
+ - `rootfs-hook`: Runs the cloud config stage `rootfs`. Notice that this runs very early in the process so things like binds or RW paths are not yet mounted.
+ - `load-config`: This parses the `/run/cos/cos-layout.env` file (usually generated by the `rootfs` stage) and loads all the configurations.
+ - `overlay-mount`: This mounts the paths set in the config (`RW_PATHS`) under the `/run/overlay` dir, so they are RW.
+ - `custom-mount`: This mounts the paths set in the config (`VOLUMES`) or in cmdline `rd.cos.mount=` in the given path (`LABEL=COS_PERSISTENT:/usr/local`)
+ - `mount-bind`: This mounts the paths set in the config (`PERSISTENT_STATE_PATHS` and `CUSTOM_BIND_MOUNTS`) as bind mounts under the `PERSISTENT_STATE_TARGET` which defaults to `/usr/local/.state`
+ - `write-fstab`: Writes the final fstab with all the mounts into `/sysroot/fstab`
+
+### UKI mode (Experimental)
+
+---
+
+There is currently support to boot in UKI mode without doing a final switch_root into `/sysroot`
+This means that the initramfs is not really an initramfs but the final system and contains all the needed stuff to boot.
```

Comment:
```suggestion
This means that the initramfs is not really an initramfs but the final system and contains all the needed parts to boot.
```

Reply:
This means that the initramfs is not really an initramfs. Nevertheless, the final system contains all the ...